lib/v2/screens/explore_screens/vote_screens/proposals/components/proposal_open_card.dart

```diff
@@ -52,111 +57,133 @@ class ProposalOpenCard extends StatelessWidget {
-                              children: [
-                                Padding(
-                                  padding: const EdgeInsets.only(right: 6.0),
-                                  child: CustomPaint(size: const Size(20, 20), painter: VotesUpArrow()),
-                                ),
-                                Flexible(
-                                  child: Text('In favour'.i18n + ': ${proposal.favourPercent}',
-                                      style: Theme.of(context).textTheme.subtitle3Green),
-                                ),
-                              ],
+                    ),
+                    Stack(
+                      children: [
+                        Padding(
+                          padding: const EdgeInsets.only(top: 25.0, left: 16.0, right: 16.0),
+                          child: ClipRRect(
+                            borderRadius: const BorderRadius.only(
+                              topLeft: Radius.circular(12),
+                              bottomLeft: Radius.circular(12),
+                              topRight: Radius.circular(12),
+                              bottomRight: Radius.circular(12),
                             ),
-                          ),
-                          Expanded(
-                            flex: 1,
-                            child: Row(
-                              mainAxisAlignment: MainAxisAlignment.center,
-                              children: [
-                                Flexible(
-                                  child: Text('Votes'.i18n + ': ${proposal.total}',
-                                      style: Theme.of(context).textTheme.subtitle3Opacity),
-                                ),
-                              ],
+                            child: StepProgressIndicator(
+                              totalSteps: proposal.total,
+                              currentStep: proposal.favour,
+                              size: 6,
+                              padding: 0,
+                              selectedColor: AppColors.green1,
+                              unselectedColor: AppColors.lightGreen6,
                             ),
                           ),
-                          Expanded(
-                            flex: 2,
-                            child: Row(
-                              mainAxisAlignment: MainAxisAlignment.end,
-                              children: [
-                                Padding(
-                                  padding: const EdgeInsets.only(right: 6.0),
-                                  child: CustomPaint(size: const Size(20, 20), painter: VotesDownArrow()),
+                        ),
+                        LayoutBuilder(
+                          builder: (_, constrains) {
+                            // Percentage to advance 0-1 scale
+                            var percent = ((proposal.voiceNeeded * 100) / proposal.total) / 100;
```

Comment:
should we move this logic to the mapper/bloc?

Reply:
fixed